pkg/account/deployer/deployer.go

```diff
@@ -17,15 +17,288 @@
+	"github.com/dynatrace/dynatrace-configuration-as-code/v2/internal/log"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/account"
-	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/manifest"
+)
+
+type (
+	AccountInfo struct {
+		Name        string
+		AccountUUID string
+	}
+	localId     = string // local (monaco related) identifier
+	envName     = string // dt environment name
+	remoteId    = string // dt entity identifier
+	projectName = string // monaco project name
+	idLookupFn  func(id localId) remoteId
 )
 
 type Options struct {
 	DryRun bool
 }
 
-// Deploy deploys all given resources to all given accounts
-func Deploy(accounts map[string]manifest.Account, resources map[string]*account.Resources, opts Options) error {
+//go:generate mockgen -source=deployer.go -destination=client_mock.go -package=deployer client
+type client interface {
+	getGlobalPolicies(ctx context.Context) (map[string]remoteId, error)
+	getManagementZones(ctx context.Context) ([]ManagementZone, error)
+	upsertPolicy(ctx context.Context, policyLevel string, policyLevelId string, policyId string, policy Policy) (remoteId, error)
+	upsertGroup(ctx context.Context, groupId string, group Group) (remoteId, error)
+	upsertUser(ctx context.Context, userId string) (remoteId, error)
+	updateAccountPolicyBindings(ctx context.Context, groupId string, policyIds []string) error
+	updateEnvironmentPolicyBindings(ctx context.Context, envName string, groupId string, policyIds []string) error
+	updateGroupBindings(ctx context.Context, userId string, groupIds []string) error
+	updatePermissions(ctx context.Context, groupId string, permissions []accountmanagement.PermissionsDto) error
+	getAccountInfo() AccountInfo
+}
+
+type AccountDeployer struct {
+	accountManagementClient client
+
+	deployedPolicies       map[localId]remoteId
+	deployedGroups         map[localId]remoteId
+	deployedMgmtZones      []accountmanagement.ManagementZoneResourceDto
+	availablePermissionIds []remoteId
+}
+
+func NewAccountDeployer(client client) *AccountDeployer {
+	return &AccountDeployer{
+		accountManagementClient: client,
+
+		deployedPolicies:       make(map[localId]remoteId),
+		deployedGroups:         make(map[localId]remoteId),
+		availablePermissionIds: make([]remoteId, 0),
+	}
+}
+
+func (d *AccountDeployer) Deploy(resources map[projectName]*account.Resources, opts Options) error {
+	if opts.DryRun {
+		log.Info("Dry run not yet implemented... nothing to do...")
+		return nil
+	}
+
+	var err error
+	d.deployedPolicies, err = d.accountManagementClient.getGlobalPolicies(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	d.deployedMgmtZones, err = d.accountManagementClient.getManagementZones(context.TODO())
+	if err != nil {
+		return err
+	}
+
+	for _, res := range resources {
+		for _, policy := range res.Policies {
+			log.Info("Deploying policy %s to account %s...", policy.Name, d.accountManagementClient.getAccountInfo().Name)
```

Comment:
I agree with the cognitive complexity linter, IMO this would be easier to read if you extracted either `deployPolicies(res.Policies)` or `deployPolicy(policy)` (no messing with the ID/uuid map then) functions (same for all types of course).

Reply:
yes, just thought calling am method and putting the returned id into a map isn't that of a complexity tbh. but i can extract it int methods . no problem